### PR TITLE
fix: MD037 space inside emphasis

### PR DESCRIPTION
**Pull request by nschonni**
Opening as draft, as this depends on https://github.com/mdn/content/pull/20113 for the other part of the flagged issues

### CODE REVIEW

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Sounds to me like a bug in Markdownlint? It should detect that it is inside HTML and that Markdown is forbidden.

Has this problem been reported? If so, what was the outcome?

Is there another workaround? If we use a Markdown table and MD mark-up, will these problems disappear? if so, I think we should do this. (Not every table can be converted to a Markdown table, so this may be a bit of work).

**Comment by nschonni** [comment]
@teoli2003 there is no special handling for HTML in Markdownlint, and the recommendation is to use the disable directives when you run into edge cases

Some of those tables could probably be converted to Markdown, and those disable directives would go away with the conversion. That's why I left them at the top of the table

**Review by caugner — APPROVED** [review]
Config change LGTM

**Comment by nschonni** [comment]
Rebased now that the license cleanup PR landed

**Comment by teoli2003** [comment]
I still believe it is a bug in Markdownlint. It should not flag Markdown errors in content of <table> or <math> as  Markdown syntax is forbidden in these elements.

I don't think it is a good idea to add 100s of magic comments to work around such a bug.

**Comment by nschonni** [comment]
> I don't think it is a good idea to add 100s of magic comments to work around such a bug.

There are 5 disables in this one, and I think the other PR that I abandoned had 40. The comments are there for when this situations happen. The same will occur with the Prettier formatting

**Comment by teoli2003** [comment]
There is no consensus to add Markdown or Prettier _magic comments_ on the pages. For the moment we add/activate rules that don't need these.

Additional question: Is MD037 auto-fixable? The Markdownlint workflow is _informative_: it tells authors and reviewers that the Markdownlint fixing workflow, running once a day, will fix something, but doesn't force authors or reviewers to act (that's why the annotation are all _warnings_).

**Comment by nschonni** [comment]
Yes, this rule is fixable https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md037---spaces-inside-emphasis-markers and the docs also talk about why this one is a functional issue. Spaces around the emphasis marker mean Markdown no longer treats the block as emphasised

File: files/en-us/web/html/global_attributes/inputmode/index.md [hunk 1]
```diff
@@ -43,8 +43,8 @@ The attribute can have any of the following values:
   - : Numeric input keyboard, but only requires the digits 0–9.
     Devices may or may not show a minus key.
 - `tel`
-  - : A telephone keypad input, including the digits 0–9, the asterisk (<kbd>*</kbd>), and the pound (<kbd>#</kbd>) key.
-    Inputs that *require\* a telephone number should typically use `{{HTMLElement("input/tel", '&lt;input type="tel"&gt;')}}` instead.
+  - : A telephone keypad input, including the digits 0–9, the asterisk (<kbd>\*</kbd>), and the pound (<kbd>#</kbd>) key.
+    Inputs that *require* a telephone number should typically use `{{HTMLElement("input/tel", '&lt;input type="tel"&gt;')}}` instead.
```
Comment (Josh-Cena) [inline files/en-us/web/html/global_attributes/inputmode/index.md:47]:
```suggestion
    Inputs that _require_ a telephone number should typically use [`<input type="tel">`](/en-US/docs/Web/HTML/Element/input/tel) instead.
```

**Review by Josh-Cena — COMMENTED** [review]
@nschonni Can we get the two valid fixes in without enabling the rule? It seems to have caught far more false positives than actual mistakes. It's also unlikely that it happens a lot, considering it results in different markup.

**Comment by nschonni** [comment]
@Josh-Cena I'm not inclinced to change this because you and @teoli2003 don't like disable comments. You appear to be the only ones that keep raising this on these PRs

**Comment by Josh-Cena** [comment]
We seem to be the only ones reviewing these PRs anyway 😄 We can definitely talk with others about this and how we should proceed, but in general I don't like how this rule introduces far more disable comments than valid fixes. It's better if we can just get the fixes in.

File: .markdownlint-cli2.jsonc [hunk 2]
```diff
@@ -101,8 +101,6 @@
       ]
     },
     "MD034": false,
-    // Pending https://github.com/mdn/content/pull/20115
-    "MD037": false,
     "MD040": false,
```
Comment (wbamberg) [inline .markdownlint-cli2.jsonc:104]:
```suggestion
    "MD037": false,
    "MD040": false,
```

File: files/en-us/learn/getting_started_with_the_web/javascript_basics/index.md [hunk 3]
```diff
@@ -190,6 +190,7 @@ If your comment contains no line breaks, it's an option to put it behind two sla
 
 An `{{Glossary("operator")}}` is a mathematical symbol that produces a result based on two values (or variables). In the following table, you can see some of the simplest operators, along with some examples to try in the JavaScript console.
 
+<!-- markdownlint-disable MD037 -->
```
Comment (wbamberg) [inline files/en-us/learn/getting_started_with_the_web/javascript_basics/index.md:193]:
```suggestion
```

File: files/en-us/learn/javascript/first_steps/math/index.md [hunk 4]
```diff
@@ -115,6 +115,7 @@ The result is then 77, as initially expected.
 
 Arithmetic operators are the basic operators that we use to do sums in JavaScript:
 
+<!-- markdownlint-disable MD037 -->
```
Comment (wbamberg) [inline files/en-us/learn/javascript/first_steps/math/index.md:118]:
```suggestion
```

File: files/en-us/web/api/web_audio_api/basic_concepts_behind_web_audio_api/index.md [hunk 5]
```diff
@@ -131,6 +131,7 @@ Each audio buffer may contain different numbers of channels. Most modern audio d
 
 When the numbers of channels of the input and the output don't match, up-mixing, or down-mixing, must be done. The following rules, controlled by setting the {{domxref("AudioNode.channelInterpretation")}} property to `speakers` or `discrete`, apply:
 
+<!-- markdownlint-disable MD037 -->
```
Comment (wbamberg) [inline files/en-us/web/api/web_audio_api/basic_concepts_behind_web_audio_api/index.md:134]:
```suggestion
```

File: files/en-us/web/api/webgl_compressed_texture_astc/index.md [hunk 6]
```diff
@@ -29,6 +29,7 @@ This extension exposes one new methods.
 
 The compressed texture formats are exposed by 28 constants and can be used in two functions: {{domxref("WebGLRenderingContext.compressedTexImage2D", "compressedTexImage2D()")}} and {{domxref("WebGLRenderingContext.compressedTexSubImage2D", "compressedTexSubImage2D()")}}.
 
+<!-- markdownlint-disable MD037 -->
```
Comment (wbamberg) [inline files/en-us/web/api/webgl_compressed_texture_astc/index.md:32]:
```suggestion
```

File: files/en-us/web/javascript/guide/expressions_and_operators/index.md [hunk 7]
```diff
@@ -452,6 +452,7 @@ These operators work as they do in most other programming languages when used wi
 
 In addition to the standard arithmetic operations (`+`, `-`, `*`, `/`), JavaScript provides the arithmetic operators listed in the following table:
 
+<!-- markdownlint-disable MD037 -->
```
Comment (wbamberg) [inline files/en-us/web/javascript/guide/expressions_and_operators/index.md:455]:
```suggestion
```

**Comment by wbamberg** [comment]
@Josh-Cena , as discussed, I've removed the enabling of MD037 and the associated directives, and kept the other changes in this PR. Do you want to take a look?

**Comment by nschonni** [comment]
@wbamberg if you wanted those changes, you should have created a new PR, not hijacked mine

**Pull request closed without merging**